### PR TITLE
search: filter results by last updated date

### LINKED ISSUE #13: Show & filter by last updated in `search` results
I like to use `search -o updated` to search for actively maintained apps, but
currently there's no way to tell just how recently any of them is updated.

From there I'd like to request,

* Show last updated date in `search` results. `yyyy-mm-dd` or `<n> days` or `<n>d`.
* Filter apps by last updated date, e.g. `search -d 14` for last 14 days.
* Lastly, let's reverse the list so the newer come before the older.

Some of these may apply to "added", too.

### PR DESCRIPTION
**Pull request by lolilolicon**
Select apps updated today (UTC):

    fdroidcl search -d 1

Select apps NOT updated since a year ago:

    fdroidcl search -d -365

The default value `-d 0` disables this filter.

Fixes #13

---

Note the help message does not indicate a negative value is allowed or what it entails; we should fix that, I think, but the line is already too long...

Also, obviously, this is only my second go... corrections welcome!

### CODE REVIEW

File: cmd/fdroidcl/search.go [hunk 1]
```diff
@@ -190,6 +195,23 @@ func filterAppsUpdates(apps []fdroidcl.App, device *adb.Device) []fdroidcl.App {
 	return result
 }
 
+func filterAppsAge(apps []fdroidcl.App, days int) []fdroidcl.App {
```
Comment (mvdan) [inline cmd/fdroidcl/search.go]: `Age` is a little confusing - I'd prefer `LastUpdated`.

File: cmd/fdroidcl/search.go [hunk 2]
```diff
@@ -190,6 +195,23 @@ func filterAppsUpdates(apps []fdroidcl.App, device *adb.Device) []fdroidcl.App {
...
+	var result []fdroidcl.App
+	older := false
```
Comment (mvdan) [inline cmd/fdroidcl/search.go]: I'd prefer that you flipped this and used `==` instead of `!=` below, to simplify the logic a little.

Comment (mvdan) [inline cmd/fdroidcl/search.go]: Also naming it something like `newer` instead of `older`

File: cmd/fdroidcl/search.go [hunk 3]
```diff
@@ -24,6 +25,7 @@ var (
 	quiet     = cmdSearch.Flag.Bool("q", false, "Print package names only")
 	installed = cmdSearch.Flag.Bool("i", false, "Filter installed apps")
 	updates   = cmdSearch.Flag.Bool("u", false, "Filter apps with updates")
+	days      = cmdSearch.Flag.Int("d", 0, "Filter apps last updated in the last <n> days")
```
Comment (mvdan) [inline cmd/fdroidcl/search.go]: I think adding something for negative numbers is necessary. If it turns out that `fdroidcl -h` is ugly or doesn't print well with it, that will have to be solved separately.

**Comment by mvdan** [comment]
A few nits, but well done! Happy to see more patches.

**Comment by lolilolicon** [comment]
OK I'll correct those. Not sure how to proceed though, another pull request? New commits? Rewrite history somehow? Please recommend :)

**Comment by mvdan** [comment]
Generally you update the commit, something like:

```
# do your changes
git commit -u --amend
git push -f origin filter_age
```

**Comment by mvdan** [comment]
Whoops, I broke this PR with my recent commits. If you know how, mind rebasing and adding the `len(apps) > 0)` check I added to the rest of the filters?

**Comment by lolilolicon** [comment]
Cool now?

**Comment by mvdan** [comment]
Yep! Thanks.

**Pull request merged**